airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/state/object_storage/ObjectStorageDestinationStateManager.kt

```diff
@@ -183,6 +219,15 @@ class ObjectStorageFallbackPersister(
         // Add a suffix matching an OPTIONAL -[0-9]+ ordinal
         val matcher =
             pathFactory.getPathMatcher(stream, suffixPattern = OPTIONAL_ORDINAL_SUFFIX_PATTERN)
+
+        if (!(stream.importType == Overwrite || stream.minimumGenerationId > 0)) {
```

Comment:
Can we add a comment explaining why we we can do this optimization in each of these cases?

Reply:
We should definitely include something about these calls being particularly pathological for the append case due to the number of objects we page through increasing over time.

Reply:
Also we should consider making this boolean logic an extension function to encapsulate the goofy booleaness.